staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/sample_and_watermark.go

```diff
@@ -161,14 +162,15 @@ func (saw *sampleAndWaterMarkHistograms) SetX1(x1 float64) {
 
 func (saw *sampleAndWaterMarkHistograms) innerSet(updateXOrX1 func()) {
 	var when time.Time
+	whenP := &when // tiptoe around https://github.com/golang/go/issues/43570 (for kube #97685)
 	var whenInt int64
 	var acc sampleAndWaterMarkAccumulator
 	var wellOrdered bool
 	func() {
 		saw.Lock()
 		defer saw.Unlock()
 		when = saw.clock.Now()
```

Comment:
As per the golang bug this is where `when` gets corrupted. Even if we use `whenP` for quantize, other things that use `when` (like `saw.lastSet = when` below) will still be operating on a corrupted time struct.

Reply:
See #97860

Reply:
good point. #97860 seems cleaner, at least for this file. can someone run the tests on ARM with that PR and verify the issue is resolved?

Reply:
@liggitt I have no idea how hard it would be to check, but it might be worth poking through the codebase for other places where there are anonymous functions modifying variables from the outer context.

Reply:
a quick sweep of anonymous functions only turned up these:

https://github.com/kubernetes/kubernetes/blob/bde721813437d22206b59e4bbe0f8dca2b5209e4/pkg/kubelet/status/status_manager.go#L516-L523

https://github.com/kubernetes/kubernetes/blob/bde721813437d22206b59e4bbe0f8dca2b5209e4/staging/src/k8s.io/apiserver/pkg/server/hooks.go#L198

https://github.com/kubernetes/kubernetes/blob/bde721813437d22206b59e4bbe0f8dca2b5209e4/staging/src/k8s.io/apiserver/pkg/server/hooks.go#L212

https://github.com/kubernetes/kubernetes/blob/bde721813437d22206b59e4bbe0f8dca2b5209e4/staging/src/k8s.io/cloud-provider/controllers/service/controller.go#L733

however, none of those changed since 1.19.x:

https://github.com/kubernetes/kubernetes/blob/85c99cf988c9675e83361034fb06db0b4d47a318/pkg/kubelet/status/status_manager.go#L516-L523

https://github.com/kubernetes/kubernetes/blob/85c99cf988c9675e83361034fb06db0b4d47a318/staging/src/k8s.io/apiserver/pkg/server/hooks.go#L198

https://github.com/kubernetes/kubernetes/blob/85c99cf988c9675e83361034fb06db0b4d47a318/staging/src/k8s.io/apiserver/pkg/server/hooks.go#L212

https://github.com/kubernetes/kubernetes/blob/85c99cf988c9675e83361034fb06db0b4d47a318/staging/src/k8s.io/cloud-provider/controllers/service/controller.go#L733

so I'd be more inclined to leave those alone and wait for the golang fix for those